src/storage/mod.rs

```diff
@@ -32,6 +69,12 @@ pub(crate) struct Blob {
     pub(crate) compression: Option<CompressionAlgorithm>,
 }
 
+impl Blob {
+    pub(crate) fn is_empty(&self) -> bool {
+        self.mime == "application/x-empty"
```

Comment:
This seems wrong - shouldn't it be `self.content.len() == 0` instead? I don't trust the remote not to return `application/octet-stream`.

Reply:
I moved that from the `File`  impl in `web::file`: 

https://github.com/rust-lang/docs.rs/blob/b2eabd65353f883280b89c7cd7d521857419dab3/src/web/file.rs#L47-L49

But I'm happy to change it here.

Reply:
Ah ok, this is fine if it's pre-existing.